arch/arm/soc/st_stm32/stm32f4/arm_mpu_mem_cfg.h

```diff
@@ -20,6 +20,8 @@
 #define REGION_FLASH_SIZE REGION_512K
 #elif CONFIG_FLASH_SIZE == 1024
 #define REGION_FLASH_SIZE REGION_1M
+#elif CONFIG_FLASH_SIZE == 1536
+#define REGION_FLASH_SIZE REGION_1M
```

Comment:
What about setting 2M here?
Looking to memory map(cf MCU datasheet), beyond flash, there is a "reserved zone" (0x0818 000 - 0x0FFF FFFF). This reserved zone is actually not addressed by the bus (from a HW point of view) and accesses
will result to abort.
If we set MPU flash zone to 2M, 0.5M will be set in this zone, and no accesses will be possible.
We keep the functionality, maybe one can confirm, but I think there is no loss of functionality, MPU wise.
Using 1M, we loose MPU functionality on last 0.5M part.